umake/ui/cli/__init__.py

```diff
@@ -157,6 +158,65 @@ def mangle_args_for_default_framework(args):
     return result_args
 
 
+def get_frameworks_list(args):
+    """
+    Get a frameworks list based on the arguments.
+    Multiple forms of the frameworks list can ge given:
+        - List with all frameworks
+        - List with just only installed frameworks
+        - List with just installable frameworks
+
+    Args:
+        args: It's the return of parse_args function. Just args.list, args.list_available and args.list_installed
+        are need to this function.
+
+    Return:
+        Give the frameworks list as a string
+    """
+    categories = list_frameworks()
+    print_result = str()
+
+    if args.list or args.list_available:
+        # Sort the categories to prevent a random list at each new program execution
+        for category in sorted(categories, key=lambda cat: cat["category_name"]):
+            if category["is_installed"] == 0:
```

Comment:
Use the constants (see previous comment) directly here rather than reharcoding the enum. Imagine we add another new state in the middle, this code would need to adapt. If we use the constants, no need! :)

On a side-note, I don't think that the fact that a category is installed or not or partionnaly is of any value for users. For instance, the "ide" category will ever be fully installed or you will be driven insane ;)